Structure_Engine/Create/SectionProperty.cs

```diff
@@ -0,0 +1,121 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Structure.SectionProperties;
+using BH.oM.Geometry.ShapeProfiles;
+using BH.oM.Structure.MaterialFragments;
+using BH.oM.Geometry;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates a Section property from a Profile and a Material. The type of section that will be created depends on the material provided. Null material or unsupported materials will return a GenericSection.")]
+        [Input("profile", "The profile of the section property")]
+        [Input("material", "The material of the section property. Used to determain which type of section that will be created. If null or a not yet explicitly supported material type, a generic section will be created.")]
+        [Input("name", "The name of the section property")]
+        [Output("section", "The created section property of a type matching the material provided.")]
+        public static IGeometricalSection SectionPropertyFromProfile(IProfile profile, IMaterialFragment material = null, string name = "")
+        {
+            MaterialType materialType = material == null ? MaterialType.Undefined : material.IMaterialType();
+
+            switch (materialType)
+            {
+                case MaterialType.Steel:
+                    return SteelSectionFromProfile(profile, material as Steel, name);
+                case MaterialType.Concrete:
+                    return ConcreteSectionFromProfile(profile, material as Concrete, name);
+                case MaterialType.Aluminium:
+                    return AluminiumSectionFromProfile(profile, material as Aluminium, name);
+                case MaterialType.Timber:
+                    return TimberSectionFromProfile(profile, material as Timber, name);
+                case MaterialType.Rebar:
+                case MaterialType.Tendon:
+                case MaterialType.Glass:
+                case MaterialType.Cable:
+                case MaterialType.Undefined:
```

Comment:
Should generic materials, with `MaterialType.Undefined` also fall through to raise the warning?

Reply:
Hmm you might be right! Kind of wanted to keep warnings for whenever GenericSection was used as a highlight that you might want to do something else..

Reply:
Yes, thought about it as well, but in general I thought it was good to keep the warning as a reminder that you might want to use something else. Can remove later if to annoying?